### PR TITLE
SimTrack to CSC digi and stub matching in CSC Validation code

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

This PR builds on top of https://github.com/cms-sw/cmssw/pull/26249 and https://github.com/cms-sw/cmssw/pull/29042. This code matches CSC wire/strip/comparator digis and ALCT/CLCT/LCT stubs to SimTracks. 

Tao Huang and I have been developing this validation code since ~2013 and have been using it in our trigger studies for the GEM TDR, Muon Phase-II TP, Muon Upgrade TDR. Most recently in the L1T TDR (https://cms-docdb.cern.ch/cgi-bin/DocDB/RetrieveFile?docid=13929&filename=TDR-20-001-paper-v1.pdf&version=4, section 2.5.5).

#### PR validation:

The code was tested with WF 22034.0.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

FYI @tahuang1991

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
@dildick You are not enabling the code in any py config file, so no plots are produced, is this intentional? Thanks

**Comment by dildick** [comment]
@jfernan2 Yes, for now I'm not adding plots to the validation package yet.

**Comment by dildick** [comment]
In a next PR, I want to add efficiency plots for wire digi, comparator digi, ALCT, CLCT and LCT. Not all plots necessarily have to be made for each validation. Some - perhaps even most - should be made for expert studies for the CSC local trigger emulator.

**Comment by jfernan2** [comment]
+1

**Comment by jfernan2** [comment]
+1

File: Validation/MuonCSCDigis/BuildFile.xml [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 <use   name="DataFormats/Common"/>
+<use   name="DataFormats/GEMDigi"/>
+<use   name="DataFormats/MuonDetId"/>
+<use   name="DQMServices/Core"/>
 <use   name="FWCore/Framework"/>
 <use   name="FWCore/MessageLogger"/>
 <use   name="FWCore/ParameterSet"/>
 <use   name="FWCore/ServiceRegistry"/>
 <use   name="Geometry/CSCGeometry"/>
+<use   name="Geometry/GEMGeometry"/>
```
Comment (silviodonato) [inline Validation/MuonCSCDigis/BuildFile.xml]: Could you check if DataFormats/GEMDigi, Geometry/GEMGeometry, DataFormats/MuonDetId are really needed?

Comment (dildick) [reply Validation/MuonCSCDigis/BuildFile.xml]: It would appear they are not.

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1
No changes as expected

**Comment by silviodonato** [comment]
please test workflow 23234.1001, 23434.1001

**Comment by silviodonato** [comment]
+1

**Pull request merged**